tools/c_ast/src/lib.rs

```diff
@@ -0,0 +1,204 @@
+    {
+        match storage_class.as_ref().map(|s| s.as_str()) {
+            None if name == "main" => *annotation = Some(FunctionAnnotation::Entry),
+            Some("static") => *annotation = Some(FunctionAnnotation::Static),
+            _ => {}
+        }
+    }
+
+    for node in ast.inner.iter_mut() {
+        annotate_ast(node);
+    }
+}
+
+#[derive(Serialize)]
+pub struct ClangAst {
+    pub asts: HashMap<String, Node<Clang>>,
+    pub source_representation: Id,
+}
+
+impl std::fmt::Display for ClangAst {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "C ASTs for {:?}", self.asts.keys())
+    }
+}
+
+impl Representation for ClangAst {
+    fn name(&self) -> &'static str {
+        "clang_ast"
+    }
+
+    fn materialize(&self, path: &std::path::Path) -> std::io::Result<()> {
+        let file = std::fs::File::create(path)?;
+        serde_json::to_writer(file, self).map_err(Into::into)
+    }
+}
+
+pub struct ParseToAst;
+
+impl Tool for ParseToAst {
+    fn name(&self) -> &'static str {
+        "parse_to_ast"
+    }
+
+    fn might_write(&mut self, context: MightWriteContext) -> MightWriteOutcome {
+        if context
+            .ir
+            .get_by_representation::<full_source::RawSource>()
+            .next()
+            .is_some()
+        {
+            MightWriteOutcome::Runnable([].into())
+        } else {
+            MightWriteOutcome::TryAgain
+        }
+    }
+
+    fn run(self: Box<Self>, context: RunContext) -> Result<(), Box<dyn std::error::Error>> {
+        for (id, rs) in context
+            .ir_snapshot
+            .get_by_representation::<full_source::RawSource>()
+        {
+            let working_dir = tempfile::TempDir::new()?;
+
+            let src_dir = tempfile::TempDir::new()?;
+            rs.dir.materialize(src_dir.path())?;
+            let src_dir_prefix = format!("{}/", src_dir.path().to_str().unwrap());
+
+            Command::new("cmake")
+                .args(["-DCMAKE_EXPORT_COMPILE_COMMANDS=1"])
+                .arg("-S")
+                .arg(src_dir.path())
+                .arg("-B")
+                .arg(working_dir.path())
+                .output()?;
+
+            #[derive(Deserialize, Debug)]
+            struct CompileCommand {
+                command: String,
+                file: String,
+            }
+            let ccs: Vec<CompileCommand> = serde_json::de::from_reader(File::open(
+                working_dir.path().join("compile_commands.json"),
+            )?)?;
+
+            let mut asts: HashMap<String, clang_ast::Node<Clang>> = Default::default();
+
+            info!(
+                "Parsing {} files: {}",
+                ccs.len(),
+                ccs.iter()
+                    .map(|cc| cc.file.as_str())
+                    .collect::<Vec<&str>>()
+                    .join(", ")
+            );
+
+            for cc in ccs {
+                let file = cc.file.replace(src_dir_prefix.as_str(), "");
```

Comment:
Should this be a call to `str::strip_prefix` instead? It looks like you only want this to match once at the start.